### ISSUE TITLE
False-positive MagicNumber when using ranges

### ISSUE THREAD

**Issue by erdlet** [arturbosch/detekt#1368]
## Expected Behavior
A range like ` internal val SOME_RANGE = 0..7` should not be reported as MagicNumber,
as it is stored as extra value and not somewhere in the code.

## Observed Behavior
A range like the example above is reported as MagicNumber, whereas a range like `val OTHER_RANGE=0..2` is not.

## Steps to Reproduce
1) Configure a Gradle build with detekt
2) Create some Kotlin file with two ranges in it (where one needs a end with a value greater than 2)
3) Run detekt. The range with the higher end will be reported

## Context
The extraction of the range should provide a clean specification of substring ranges during the construction of a complex string.

## Your Environment
* Version of detekt used: 1.0.0-RC11
* Version of Gradle used (if applicable): 4.8
* Operating System and version: Fedora 29

**Comment by arturbosch** [comment]
I think on this one we can argue. I wouldn't call it false positives but rather something that can/should be configurable?

**Comment by schalkms** [comment]
I don't think this rule should make a difference between a magic number in a range or elsewhere.
Personally, I think that `4..41` is also *magic*.

**Comment by erdlet** [comment]
@arturbosch For me it would be OK if this MagicNumber rule could be configured, so I can deactivate it for cases like the one above.

@schalkms For inline ranges which appear e.g. in a substring like `"test".substring(0..2)` I fully agree with you, that is really magic. But in case I assign a range to a constant which gives that range an explicit context that explains the usage, there is no magic in my eyes.

**Comment by arturbosch** [comment]
So a config option for number literal ranges sounds good to me ?

**Comment by glebsts** [comment]
When I dive into unknown code, that wouldn't make sense for me:
`val VALID_RESPONSE_CODE_RANGE = 13 .. 27`
while this would make much more sense
```
const MIN_VALID_RESPONSE_CODE_VALUE = 13
const MAX_VALID_RESPONSE_CODE_VALUE = 27
val VALID_RESPONSE_CODE_RANGE = MIN_VALID_RESPONSE_VALUE .. MAX_VALID_RESPONSE_VALUE
```
Slightly synthetic example, but I believe you get the idea.
